### ISSUE TITLE
feature: line jump with hints on the same column

### ISSUE THREAD

**Issue by shadyalfred** [folke/flash.nvim#214]
### Did you check the docs?

- [X] I have read all the flash.nvim docs

### Is your feature request related to a problem? Please describe.

Showing the labels at the beginning of the line can be hard to follow specially when the lines are indentated and there is a big gap of space between the label hint and the start of the text in the line. 

### Describe the solution you'd like

Something like this
![image](https://github.com/folke/flash.nvim/assets/3685582/83930f69-8811-413d-9227-bfbaec006af5)

**Comment by IndianBoy42** [comment]
In the examople code for line jumps `  label = { after = { 0, 0 } },` the tuple is the offset from the match to show the label, can you try changing the first `0` based on `vim.api.nvim_win_get_cursor` to get it to render at the same column as the cursor?

**Comment by shadyalfred** [comment]
Good idea, @IndianBoy42!

```lua
local col = vim.api.nvim_win_get_cursor(0)[2]

flash.jump({
  ...
  label = {
    before = { 0, col },
  },
})
```
![image](https://github.com/folke/flash.nvim/assets/3685582/901c50a0-e955-46ae-82d4-92d9c32e3a7b)

The label is shown in the correct position, but jumping to it moves the cursor to the beginning of that line which is a bit disorienting.

**Comment by shadyalfred** [comment]
This works great!

```lua
local col = vim.api.nvim_win_get_cursor(0)[2]

flash.jump({
  ...
  label = {
    before = { 0, col },
  },
})

vim.api.nvim_input(col .. 'l')
```

**Comment by IndianBoy42** [comment]
https://github.com/folke/flash.nvim/blob/967117690bd677cb7b6a87f0bc0077d2c0be3a27/README.md?plain=1#L160

You can also use `opts.jump.offset`

**Comment by shadyalfred** [comment]
Yes, I've found out about it after closing the issue. Also disabling the search highlight because it was confusing.
```lua
-- Move to lines beneath the cursor
local col = vim.api.nvim_win_get_cursor(0)[2]

flash.jump({
  jump = {
    offset = col,
  },
  search = {
    mode = 'search',
    max_length = 0,
    forward = true,
    wrap = false,
    multi_window = false,
  },
  pattern = '^',
  label = {
    before = { 0, col },
  },
  highlight = {
    matches = false,
  },
})
```

```lua
-- move to lines above the cursor
local col = vim.api.nvim_win_get_cursor(0)[2]

flash.jump({
  jump = {
    offset = col,
  },
  search = {
    mode = 'search',
    max_length = 0,
    forward = false,
    wrap = false,
    multi_window = false,
  },
  pattern = '^',
  label = {
    before = { 0, col },
  },
  highlight = {
    matches = false,
  },
})
```
